pokete_classes/deck.py

```diff
@@ -14,73 +14,121 @@
 from .color import Color
 from .poke import Poke
 from .ui_elements import StdFrame2
+from .tss import tss
 
 
 class Deck(detail.Informer):
     """Deck to see Poketes in"""
 
     def __init__(self, height, width, figure, abb_funcs):
-        self.map = gm.GameMap(height, width)
+        self.map = gm.GameMap(height, width, name="deck")
         self.submap = gm.GameSubmap(self.map, 0, 0, height, width, "decksubmap")
         self.exit_label = se.Text(f"{Action.DECK.mapping}: Exit  ")
         self.move_label = se.Text(f"{Action.MOVE_POKETE.mapping}: Move    ")
         self.move_free = se.Text(f"{Action.FREE_POKETE.mapping}: Free")
         self.index = se.Object("*")
         self.figure = figure
         self.abb_funcs = abb_funcs
+        self.pokes = []
+        self.label = ""
+        self.overview = None
         # adding
         self.exit_label.add(self.submap, 0, self.submap.height - 1)
         self.move_label.add(self.submap, 9, self.submap.height - 1)
         self.move_free.add(self.submap, 20, self.submap.height - 1)
 
-    def rem_pokes(self, pokes):
+    def rem_pokes(self):
         """Removes all Poketes from the Deck
         ARGS:
             pokes: List of Pokes being removed"""
-        for poke in pokes:
+        for poke in self.pokes:
             self.remove(poke)
 
-    def __call__(self, p_len, label="Your full deck", in_fight=False):
+    def resize_view(self):
+        """Manages recursive view resizing"""
+        self.exit_label.remove()
+        self.move_label.remove()
+        self.move_free.remove()
+
+        self.submap.resize(tss.height - 1, tss.width)
+
+        self.exit_label.add(self.submap, 0, self.submap.height - 1)
+        self.move_label.add(self.submap, 9, self.submap.height - 1)
+        self.move_free.add(self.submap, 20, self.submap.height - 1)
+
+        if all(i.ico.map == self.map for i in self.pokes):
+            self.rem_pokes()
+
+        while len(self.map.obs) > 0:
+            self.map.obs[0].remove()
+        self.map.resize(5 * int((len(self.pokes) + 1) / 2) + 2, tss.width,
+                        self.map.background)
+        self.overview.resize_view()
+        se.Text(self.label, esccode=Color.thicc).add(self.map, 2, 0)
+        se.Square("|", 1, self.map.height - 2).add(self.map,
+                                                   round(self.map.width / 2),
+                                                   1)
+        StdFrame2(self.map.height - 1, self.map.width).add(self.map, 0, 0)
+        self.add_all(True, not all(i.ico.map == self.map for i in self.pokes))
+        self.index.add(
+            self.map,
+            self.pokes[self.index.index].text_name.x
+            + len(self.pokes[self.index.index].text_name.text) + 1,
+            self.pokes[self.index.index].text_name.y
+        )
+
+    def __call__(self, overview, p_len, label="Your full deck", in_fight=False):
```

Comment:
Using `__call__` feels weird, is there a particular reason this cannot be a regular method such as `open(self, ...)`?